### DISCUSSION TITLE
Unable to start a session for windows app using Appium

### DISCUSSION THREAD

**Discussion by bp24acharya** [appium/appium#18187] [General]
package automationtest;

import org.testng.annotations.AfterMethod;
import org.testng.annotations.Test;
import org.testng.annotations.BeforeMethod;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.concurrent.TimeUnit;
import org.openqa.selenium.remote.DesiredCapabilities;
import io.appium.java_client.windows.WindowsDriver;
public class AutomateWindowsApp {
public static WindowsDriver driver = null;
   @BeforeMethod
	public void setUp() {
		DesiredCapabilities cap = new DesiredCapabilities();
//windows application id needs to be given to open the app. Run Get-StartApps in powershell to get all windows app ids.
		cap.setCapability("app", " C:\\Amnexis\\HomeCareAdminApp.exe");
		//cap.setCapability("app", "C:\\Users\\bijay\\Desktop\\HomeCareAdminApp.exe");
		
		
//add platformname and device name
		cap.setCapability("platformName", "Windows");
		cap.setCapability("deviceName", "WindowsPC");
		try {
//create webdriver instance
			
		 driver = new WindowsDriver(new URL("http://127.0.0.1:4723"), cap);
		} catch (MalformedURLException e) {
			e.printStackTrace();
		}

**Error** : org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 400. Message: Bad capabilities. Specify either app or appTopLevelWindow to create a session 
Host info: host: 'DESKTOP-N7BNLUE', ip: '192.168.68.164'
Build info: version: '4.8.0', revision: '267030adea'
System info: os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '17.0.2'
Driver info: io.appium.java_client.windows.WindowsDriver
Command: [null, newSession {capabilities=[{appium:app= C:\Amnexis\HomeCareAdminApp.exe, appium:automationName=Windows, appium:deviceName=WindowsPC, platformName=WINDOWS}], desiredCapabilities=Capabilities {app:  C:\Amnexis\HomeCareAdminAp..., appium:automationName: Windows, deviceName: WindowsPC, platformName: WINDOWS}}]
Capabilities {}

**Comment by KazuCocoa** [comment]
What the Appium server log?

**Reply by bp24acharya** [reply]
Windows Application Driver listening for requests at: http://127.0.0.1:4723/
Press ENTER to exit.


==========================================
POST /session HTTP/1.1
Accept: */*
Content-Length: 269
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: appium/8.3.0 (selenium/4.8.0 (java windows))
X-Idempotency-Key: 19b59ab8-09b2-4fd4-80ec-a954ef7be4fc

{
  "capabilities": {
    "firstMatch": [
      {
      }
    ],
    "alwaysMatch": {
      "appium:app": " C:\\Amnexis\\HomeCareAdminApp.exe",
      "appium:automationName": "Windows",
      "appium:deviceName": "WindowsPC",
      "platformName": "windows"
    }
  }
}
HTTP/1.1 400 Bad Request
Content-Length: 141
Content-Type: application/json

**Reply by KazuCocoa** [reply]
Did you use https://github.com/appium/appium-windows-drive ? It seems like you used https://github.com/microsoft/WinAppDriver directly, then it was because https://github.com/microsoft/WinAppDriver did not follow W3C protocol. Please ask https://github.com/microsoft/WinAppDriver

**Comment by bp24acharya** [comment]
Hello
Thanks very much for your reply. I didn't get what you are saying. Would
you have any time to see please or let me know What I should do in this
case.
Regards

On Mon, Feb 13, 2023 at 2:25 AM Kazuaki Matsuo ***@***.***>
wrote:

> Did you use https://github.com/appium/appium-windows-drive ? It seems
> like you used https://github.com/microsoft/WinAppDriver directly, then it
> was because https://github.com/microsoft/WinAppDriver did not follow W3C
> protocol. Please ask https://github.com/microsoft/WinAppDriver
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/appium/appium/discussions/18187#discussioncomment-4952552>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQNDXVGJXZJYFVKKF74EIEDWXGLP7ANCNFSM6AAAAAAUY2CE7Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks and Regards
Prarthana Mehta Acharya
Email: ***@***.***
-------------------------------------------------

**Reply by KazuCocoa** [reply]
Could you share logs when you start Appium?
For example, when you start an Appium, you may be able to see:

```
[Appium] Welcome to Appium v2.0.0-beta.53 (REV 011810e721c4c26be997c7c0b62739c7b991526d)
[Appium] Attempting to load driver xcuitest...
...
[Appium]   - uiautomator2@2.12.3 (automationName 'UiAutomator2')
[Appium] Available plugins:
[Appium]   - execute-driver@3.0.0
[Appium]   - ocr@0.1.3
```

This is Appium 2.0, so it might be different from your local one, but such log should be there when you start an Appium.

Logs you shared in https://github.com/appium/appium/discussions/18187#discussioncomment-4947891 was probably starting https://github.com/microsoft/WinAppDriver directly, not via Appium. Then, https://github.com/microsoft/WinAppDriver is proper place for you

**Comment by bp24acharya** [comment]
Hello,
I don't start Appium when running the script for Windows APP. I am running
Winapp driver and here is the logs

My other windows app like Notepad is running well, I am not sure where the
issue is actually.


Windows Application Driver listening for requests at: http://127.0.0.1:4723/
Press ENTER to exit.


==========================================
POST /session HTTP/1.1
Accept: */*
Content-Length: 269
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: appium/8.3.0 (selenium/4.8.0 (java windows))
X-Idempotency-Key: 6a8b13de-9d77-4ffc-8779-fb8edbdaf785

{
  "capabilities": {
    "firstMatch": [
      {
      }
    ],
    "alwaysMatch": {
      "appium:app": " C:\\Amnexis\\HomeCareAdminApp.exe",
      "appium:automationName": "Windows",
      "appium:deviceName": "WindowsPC",
      "platformName": "windows"
    }
  }
}
HTTP/1.1 400 Bad Request
Content-Length: 141
Content-Type: application/json



On Mon, Feb 13, 2023 at 9:10 AM Kazuaki Matsuo ***@***.***>
wrote:

> Could you share logs when you start Appium?
> For example, when you start an Appium, you may be able to see:
>
> [Appium] Welcome to Appium v2.0.0-beta.53 (REV 011810e721c4c26be997c7c0b62739c7b991526d)
> [Appium] Attempting to load driver xcuitest...
> ...
> [Appium]   - ***@***.*** (automationName 'UiAutomator2')
> [Appium] Available plugins:
> [Appium]   - ***@***.***
> [Appium]   - ***@***.***
>
> This is Appium 2.0, so it might be different from your local one, but such
> log should be there when you start an Appium.
>
> Logs you shared in #18187 (reply in thread)
> <https://github.com/appium/appium/discussions/18187#discussioncomment-4947891>
> was probably starting https://github.com/microsoft/WinAppDriver directly,
> not via Appium
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/appium/appium/discussions/18187#discussioncomment-4955705>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQNDXVATL44FFZE7SYSU5ZLWXH3BJANCNFSM6AAAAAAUY2CE7Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks and Regards
Prarthana Mehta Acharya
Email: ***@***.***
-------------------------------------------------

**Comment by bp24acharya** [comment]
*Just for your information, this is the log for notepad that is running
well with the same setups*


Windows Application Driver listening for requests at: http://127.0.0.1:4723/
Press ENTER to exit.


==========================================
POST /session HTTP/1.1
Accept: */*
Content-Length: 269
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: appium/8.3.0 (selenium/4.8.0 (java windows))
X-Idempotency-Key: 6a8b13de-9d77-4ffc-8779-fb8edbdaf785

{
  "capabilities": {
    "firstMatch": [
      {
      }
    ],
    "alwaysMatch": {
      "appium:app": " C:\\Amnexis\\HomeCareAdminApp.exe",
      "appium:automationName": "Windows",
      "appium:deviceName": "WindowsPC",
      "platformName": "windows"
    }
  }
}
HTTP/1.1 400 Bad Request
Content-Length: 141
Content-Type: application/json




==========================================
POST /session HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 389
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "desiredCapabilities": {
    "app": "C:\\Program Files\\Windows NT\\Accessories\\wordpad.exe",
    "platformName": "Windows",
    "deviceName": "WindowsPC"
  },
  "capabilities": {
    "firstMatch": [
      {
        "appium:app": "C:\\Program Files\\Windows
NT\\Accessories\\wordpad.exe",
        "appium:deviceName": "WindowsPC",
        "platformName": "windows"
      }
    ]
  }
}
HTTP/1.1 200 OK
Content-Length: 162
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":{"app":"C:\\Program
Files\\Windows NT\\Accessories\\wordpad.exe","platformName":"Windows"}}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/timeouts HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 38
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "type": "implicit",
  "ms": 5000
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 52
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "using": "name",
  "value": "Rich Text Window"
}
HTTP/1.1 200 OK
Content-Length: 95
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":{"ELEMENT":"42.461678"}}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element/42.461678/click
HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 23
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "id": "42.461678"
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element/42.461678/value
HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 59
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "id": "42.461678",
  "value": [
    "It was good"
  ]
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 52
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "using": "name",
  "value": "Rich Text Window"
}
HTTP/1.1 200 OK
Content-Length: 95
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":{"ELEMENT":"42.461678"}}


==========================================
GET /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element/42.461678/text
HTTP/1.1
Accept-Encoding: gzip
Cache-Control: no-cache
Connection: Keep-Alive
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)


HTTP/1.1 200 OK
Content-Length: 85
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":"It
was good"}

==========================================

POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 41
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "using": "name",
  "value": "Lists"
}
HTTP/1.1 200 OK
Content-Length: 122
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":{"ELEMENT":"84e20b37-4573-4116-98d2-68bb188f63a2"}}


==========================================
POST
/session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element/84e20b37-4573-4116-98d2-68bb188f63a2/click
HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 50
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "id": "84e20b37-4573-4116-98d2-68bb188f63a2"
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element/42.461678/value
HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 56
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "id": "42.461678",
  "value": [
    "love uuu"
  ]
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 40
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "using": "name",
  "value": "Bold"
}
HTTP/1.1 200 OK
Content-Length: 122
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":{"ELEMENT":"0d69b856-1b67-44d2-914f-7e144f2622ef"}}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 42
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "using": "name",
  "value": "Italic"
}
HTTP/1.1 200 OK
Content-Length: 122
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":{"ELEMENT":"cb09bbfd-774c-4467-9581-e9522fab254d"}}


==========================================
POST
/session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element/cb09bbfd-774c-4467-9581-e9522fab254d/click
HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 50
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "id": "cb09bbfd-774c-4467-9581-e9522fab254d"
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0}


==========================================
POST /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 45
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "using": "name",
  "value": "Underline"
}
HTTP/1.1 200 OK
Content-Length: 122
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0,"value":{"ELEMENT":"35e4be5e-9d44-4ceb-9eda-eadd905fe438"}}


==========================================
POST
/session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767/element/35e4be5e-9d44-4ceb-9eda-eadd905fe438/click
HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 50
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "id": "35e4be5e-9d44-4ceb-9eda-eadd905fe438"
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"B3BD68E0-66D4-4C57-A9DD-404D8B53E767","status":0}


==========================================
DELETE /session/B3BD68E0-66D4-4C57-A9DD-404D8B53E767 HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 0
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)


HTTP/1.1 200 OK
Content-Length: 12
Content-Type: application/json

{"status":0}


==========================================
POST /session HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 389
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "desiredCapabilities": {
    "app": "C:\\Program Files\\Windows NT\\Accessories\\wordpad.exe",
    "platformName": "Windows",
    "deviceName": "WindowsPC"
  },
  "capabilities": {
    "firstMatch": [
      {
        "appium:app": "C:\\Program Files\\Windows
NT\\Accessories\\wordpad.exe",
        "appium:deviceName": "WindowsPC",
        "platformName": "windows"
      }
    ]
  }
}
HTTP/1.1 200 OK
Content-Length: 162
Content-Type: application/json

{"sessionId":"E7DFBF86-26E2-4175-8A14-976E8D137A7A","status":0,"value":{"app":"C:\\Program
Files\\Windows NT\\Accessories\\wordpad.exe","platformName":"Windows"}}

==========================================

POST /session/E7DFBF86-26E2-4175-8A14-976E8D137A7A/timeouts HTTP/1.1
Accept-Encoding: gzip
Connection: Keep-Alive
Content-Length: 38
Content-Type: application/json; charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.59 (java windows)

{
  "type": "implicit",
  "ms": 5000
}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"E7DFBF86-26E2-4175-8A14-976E8D137A7A","status":0}

On Mon, Feb 13, 2023 at 9:59 AM Prarthana Acharya ***@***.***>
wrote:

> Hello,
> I don't start Appium when running the script for Windows APP. I am running
> Winapp driver and here is the logs
>
> My other windows app like Notepad is running well, I am not sure where the
> issue is actually.
>
>
> Windows Application Driver listening for requests at:
> http://127.0.0.1:4723/
> Press ENTER to exit.
>
>
> ==========================================
> POST /session HTTP/1.1
> Accept: */*
> Content-Length: 269
> Content-Type: application/json; charset=utf-8
> Host: 127.0.0.1:4723
> User-Agent: appium/8.3.0 (selenium/4.8.0 (java windows))
> X-Idempotency-Key: 6a8b13de-9d77-4ffc-8779-fb8edbdaf785
>
> {
>   "capabilities": {
>     "firstMatch": [
>       {
>       }
>     ],
>     "alwaysMatch": {
>       "appium:app": " C:\\Amnexis\\HomeCareAdminApp.exe",
>       "appium:automationName": "Windows",
>       "appium:deviceName": "WindowsPC",
>       "platformName": "windows"
>     }
>   }
> }
> HTTP/1.1 400 Bad Request
> Content-Length: 141
> Content-Type: application/json
>
>
>
> On Mon, Feb 13, 2023 at 9:10 AM Kazuaki Matsuo ***@***.***>
> wrote:
>
>> Could you share logs when you start Appium?
>> For example, when you start an Appium, you may be able to see:
>>
>> [Appium] Welcome to Appium v2.0.0-beta.53 (REV 011810e721c4c26be997c7c0b62739c7b991526d)
>> [Appium] Attempting to load driver xcuitest...
>> ...
>> [Appium]   - ***@***.*** (automationName 'UiAutomator2')
>> [Appium] Available plugins:
>> [Appium]   - ***@***.***
>> [Appium]   - ***@***.***
>>
>> This is Appium 2.0, so it might be different from your local one, but
>> such log should be there when you start an Appium.
>>
>> Logs you shared in #18187 (reply in thread)
>> <https://github.com/appium/appium/discussions/18187#discussioncomment-4947891>
>> was probably starting https://github.com/microsoft/WinAppDriver
>> directly, not via Appium
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/appium/appium/discussions/18187#discussioncomment-4955705>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQNDXVATL44FFZE7SYSU5ZLWXH3BJANCNFSM6AAAAAAUY2CE7Q>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Thanks and Regards
> Prarthana Mehta Acharya
> Email: ***@***.***
> -------------------------------------------------
>
>

-- 
Thanks and Regards
Prarthana Mehta Acharya
Email: ***@***.***
-------------------------------------------------

**Reply by KazuCocoa** [reply]
Then, this Appium repository is the wrong place to report. https://github.com/microsoft/WinAppDriver/issues/988 is the root cause for you.
You might be able to start a session with https://github.com/appium/appium-windows-driver, which is Appium.

**Comment by DeLaphante** [comment]
Struggling to design or build an advanced, stable and well structured test automation framework?

Check out & play around with my powerful and robust Selenium Web UI & API test automation framework on Github -> https://github.com/DeLaphante/CynkyAutomation

Save yourself some time and use this ready made framework for any project and get the technical support you need to succeed!
